### DISCUSSION TITLE
(NixOS) hyprland hasn't built since hyprcursors was introduced

### DISCUSSION THREAD

**Discussion by zachcoyle** [hyprwm/Hyprland#5083] [General]
There's a very good chance I'm doing something dumb, but ever since hyprcursors was added to the hyprland flake, I've been getting an error building hyprcursors. I can build it fine with `nix build github:hyprwm/hyprcursors` and `nix build github:hyprwm/Hyprland` but Whenever I try to build it in my flake it's failed (and I have tried EVERY permutation of `follows`). I'm hoping I've just overlooked something dumb.

```error: builder for '/nix/store/x4n8v1czay1ms00vxjazx33l42kf1zby-hyprcursor-0.pre+date=2024-03-10_05db7df.drv' failed with exit code 2;
       last 10 log lines:
       > /build/w4wl33hjpk2d639pr0mfq25h6b8gcsli-source/libhyprcursor/hyprcursor.cpp: In member function 'std::optional<std::__cxx11::basic_string<char> > CHyprcursorImplementation::loadTheme()':
       > /build/w4wl33hjpk2d639pr0mfq25h6b8gcsli-source/libhyprcursor/hyprcursor.cpp:572:109: error: 'Hyprlang::SConfigOptions' has no non-static data member named 'pathIsStream'
       >   572 |             meta = std::make_unique<Hyprlang::CConfig>(buffer, Hyprlang::SConfigOptions{.pathIsStream = true});
       >       |                                                                                                             ^
       > make[2]: *** [CMakeFiles/hyprcursor.dir/build.make:76: CMakeFiles/hyprcursor.dir/libhyprcursor/hyprcursor.cpp.o] Error 1
       > make[1]: *** [CMakeFiles/Makefile2:105: CMakeFiles/hyprcursor.dir/all] Error 2
       > make[1]: *** Waiting for unfinished jobs....
       > [ 44%] Linking CXX executable hyprcursor-util
       > [ 44%] Built target hyprcursor-util
       > make: *** [Makefile:136: all] Error 2
       For full logs, run 'nix log /nix/store/x4n8v1czay1ms00vxjazx33l42kf1zby-hyprcursor-0.pre+date=2024-03-10_05db7df.drv'.
error: 1 dependencies of derivation '/nix/store/knxcfm61m1jsx1sv2r0jp6fjlg0s72la-hyprland-0.36.0+date=2024-03-12_5da9591.drv' failed to build
error: 1 dependencies of derivation '/nix/store/73cpxwbabl01y3v0q90q86j95kqa4avp-activation-script.drv' failed to build
error: 1 dependencies of derivation '/nix/store/grpvzs85a3sh7mmj5v97rbzwcnd9r708-desktops.drv' failed to build
error: 1 dependencies of derivation '/nix/store/iwqhbd4b26lxqzd8nw8whfr33bicz0f0-home-manager-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/m4i6hacaa5ms9linnq8l99sc5x6jx432-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/pdghg31shgvbijqlicxldmbvxzlspy8s-xdg-desktop-portal-hyprland-1.3.1+date=2024-03-01_2d2fb54.drv' failed to build
error: 1 dependencies of derivation '/nix/store/i5hrwpx9wjg7m88ij9gncfp92iws6dyq-nixos-system-nixos-laptop-24.05.20240312.7eeacec.drv' failed to build```

full config is at github.com/zachcoyle/nix-config branch update-hyprland

halp @fufexan :)

**Comment by zachcoyle** [comment]
btw congrats @vaxerski you've made me a twm person

**Comment by fufexan** [comment]
Hi Zach! I'm not sure what the issue is.

I've tried building hyprland with the `follows` you have in your flake, but it did not give any errors.
`nix build github:hyprwm/Hyprland/3e930a568a76a810dde6bcabcf737a81850e7153 --override-input nixpkgs github:NixOS/nixpkgs/db339f1706f555794b71aa4eb26a5a240fb6a599 --override-input systems github:nix-systems/default-linux/31732fcf5e8fea42e59c2488ad31a0e651500f68`.

Can you try to update Hyprland?

EDIT: also tried to build your config with `nix build github:zachcoyle/nix-config/update-hyprland#nixosConfigurations.nixos-desktop.config.system.build.toplevel -L` but the error I got was related to TabNine.

**Comment by zachcoyle** [comment]
Thanks for looking at it Mihai - I ended up figuring out what my problem was (it did end up being something dumb on my end):

![20240313_11h05m18s_grim](https://github.com/hyprwm/Hyprland/assets/908716/b2bc650b-6888-481b-88e8-331072699d1a)

💀 💀 💀 

When I get a bit of time, I'll make some notes in the wiki

**Comment by zachcoyle** [comment]
Also - it's worth noting that having the overlays in the incorrect order in my case was also causing hyprwm/hyprland-plugins to make hyprland crash on me

**Comment by zachcoyle** [comment]
thank you @fufexan

**Reply by fufexan** [reply]
You're welcome. I guess the issue is that the different flakes have different input versions locked. Ideally they'd all have the same locked versions for the common inputs. I'll try to work on that.
